packages/dapp/src/pages/Main.tsx

```diff
@@ -1,36 +1,125 @@
-import { Box, Text } from 'grommet';
+import { useState, useEffect } from 'react';
+import {
+  Box,
+  Button,
+  Form,
+  FormField,
+  Text,
+  TextInput,
+} from 'grommet';
+import { encrypt, decrypt } from '../utils/crypto';
+import {useAppState} from "../store";
 
-import { useAppState } from '../store';
-import { SignInButton, SignOutButton } from '../components/buttons/web3Modal';
-import { Account } from '../components/Account';
+export const Main = () => {
+  const { encryptionKey } = useAppState();
+  const { encryptedData, switchEncryptedData } = useAppState();
 
-// Config
-import { getNetworksNames } from '../config';
+  const [value, setValue] = useState({ name: '', email: '' });
+  const [cryptoOpInProgress, setCryptoOpInProgress] = useState(false);
 
-const allowedNetworksNames = getNetworksNames();
+  const resetData = () => {
+    setValue({ name: '', email: '' });
+  };
 
-export const Main = () => {
-  const { account, isRightNetwork } = useAppState();
+  const saveData = (nextValue: { name: string, email: string }) => {
```

Comment:
it is better to move data saving functionality to the store scope. It can be a component or a hook.

Reply:
This will be fixed in my next PR.